core/thread/test/testRWLock.cxx

```diff
@@ -0,0 +1,398 @@
+// #include "TReentrantRWLock.h"
+#include "TVirtualMutex.h"
+#include "TMutex.h"
+#include "TVirtualRWMutex.h"
+#include "ROOT/TReentrantRWLock.hxx"
+#include "ROOT/TRWSpinLock.hxx"
+
+#include "../src/TRWMutexImp.h"
```

Comment:
Sometimes people (I find myself doing so more and more often) look at the tests to see how a something without sufficient documentation behaves. I think saying this is only for the unittest should make it clearer for such users.